lib/sinatra/base.rb

```diff
@@ -487,29 +487,24 @@ def cache_control(*values)
       response['Cache-Control'] = values.join(', ') if values.any?
     end
 
-    # Set the Expires header and Cache-Control/max-age directive. Amount
+    # Set the Expires header and Cache-Control header. Amount
     # can be an integer number of seconds in the future or a Time object
     # indicating when the response should be considered "stale". The remaining
     # "values" arguments are passed to the #cache_control helper:
     #
     #   expires 500, :public, :must_revalidate
-    #   => Cache-Control: public, must-revalidate, max-age=500
+    #   => Cache-Control: public, must-revalidate
     #   => Expires: Mon, 08 Jun 2009 08:50:17 GMT
     #
     def expires(amount, *values)
-      values << {} unless values.last.kind_of?(Hash)
+      cache_control(*values)
 
       if amount.is_a? Integer
         time    = Time.now + amount.to_i
-        max_age = amount
       else
         time    = time_for amount
-        max_age = time - Time.now
       end
 
-      values.last.merge!(:max_age => max_age)
```

Comment:
Instead of removing this, I would rather see `max_age` merged only if it doesn't already exist in values. This would allow someone to potentially have different values for `Cache-Control` and `Expires`. This way someone could specify a different expiration time for http 1.0 agents. Agents that support http 1.1 will give preference to max-age.